rclcpp/include/rclcpp/subscription_factory.hpp

```diff
@@ -88,15 +88,20 @@ create_subscription_factory(
   subscription_topic_stats = nullptr
 )
 {
-  auto allocator = options.get_allocator();
+  rclcpp::SubscriptionOptionsWithAllocator<AllocatorT> options_copy = options;
+  // reset callback group as the Subscription is unnecessary to hold it
+  options_copy.callback_group.reset();
```

Comment:
why do we need to do this?

Reply:
Because I want to keep consistent behavior with other entities. 
I found other entities not to hold the callback group, refer to https://github.com/ros2/rclcpp/pull/1754#issuecomment-982423135, so I think it's better not to hold a callback group in the subscription.
(Next time I will create a new issue to confirm if it's a bug when I encounter this situation.)

Reply:
Is it a correct comment?
https://github.com/ros2/rclcpp/blob/87d754b2190edf53a3f7b9f95812e5a80313f649/rclcpp/test/rclcpp/test_memory_strategy.cpp#L361 

IMO, [This](https://github.com/ros2/rclcpp/blob/87d754b2190edf53a3f7b9f95812e5a80313f649/rclcpp/test/rclcpp/test_memory_strategy.cpp#L317) seems just for a coverage test. Not to consider whether it's suitable compared with other entities.

I am not sure if there is some special case that must hold the callback group in the subscription.

Reply:
> Because I want to make the subscription keep consistent behavior with other entities.
I found other entities not to hold the callback group, refer to #1754 (comment), so I think it's better not to hold a callback group in the subscription.

That sounds fine, but it doesn't seem to be needed in this PR (which seems to be a replacement of https://github.com/ros2/rclcpp/pull/1754).
I would rather see this done in another PR if it's not needed here.

Reply:
> I am not sure if there is some special case that must hold the callback group in the subscription.

Could you explain how this change affected that test?
I see there's a lifetime issue, but it isn't to clear to me what exactly the problem is.

Reply:
> I would rather see this done in another PR if it's not needed here.

OK. I'll create a new PR.

> Could you explain how this change affected that test?

Yes, it's my pleasure. Let me try to explain it.

```patch
 std::shared_ptr<const rcl_node_t>
 NodeBase::get_shared_rcl_node_handle() const
 {
-  return node_handle_;
+  return std::shared_ptr<const rcl_node_t>(shared_from_this(), node_handle_.get());
 }
```

After making the `NodeBase` sharable with rcl_node_t, it means the dependency chain will be `SubscriptionBase -> rcl_node_t -> NodeBase`. Even if the `Node` goes out of scope, we can also use `memory_strategy()->get_group_by_subscription(subscription, weak_groups_to_nodes)` to get the `CallbackGroup` pointer because the `callback_group` and `NodeBase` are alive. So I need to make the `callback_group` out of life to make the test `TEST_F(TestMemoryStrategy, get_group_by_subscription)` passed.

I know this patch extends a bit more lifetime of `NodeBase`.

Reply:
> Even if the Node goes out of scope, we can also use memory_strategy()->get_group_by_subscription(subscription, weak_groups_to_nodes) to get the CallbackGroup pointer because the callback_group and NodeBase are alive

IMO, the test makes no sense.
It's testing stuff that's not clearly defined.

I would just update the tests without updating anything else, I would prefer to discuss consitency with other cases in a new PR.

> OK. I'll create a new PR.

:+1: thanks